drivers/spi/spi_ll_stm32.c

```diff
@@ -73,6 +75,36 @@ LOG_MODULE_REGISTER(spi_ll_stm32);
 #endif
 #endif /* CONFIG_SOC_SERIES_STM32MP1X */
 
+static void spi_stm32_pm_policy_state_lock_get(const struct device *dev)
+{
+	if (IS_ENABLED(CONFIG_PM)) {
+		struct spi_stm32_data *data = dev->data;
+
+		if (!data->pm_policy_state_on) {
+			data->pm_policy_state_on = true;
+			pm_policy_state_lock_get(PM_STATE_SUSPEND_TO_IDLE, PM_ALL_SUBSTATES);
+			if (IS_ENABLED(CONFIG_PM_S2RAM)) {
+				pm_policy_state_lock_get(PM_STATE_SUSPEND_TO_RAM, PM_ALL_SUBSTATES);
+			}
+		}
+	}
+}
+
+static void spi_stm32_pm_policy_state_lock_put(const struct device *dev)
+{
+	if (IS_ENABLED(CONFIG_PM)) {
+		struct spi_stm32_data *data = dev->data;
+
+		if (data->pm_policy_state_on) {
+			data->pm_policy_state_on = false;
+			pm_policy_state_lock_put(PM_STATE_SUSPEND_TO_IDLE, PM_ALL_SUBSTATES);
+			if (IS_ENABLED(CONFIG_PM_S2RAM)) {
+				pm_policy_state_lock_put(PM_STATE_SUSPEND_TO_RAM, PM_ALL_SUBSTATES);
+			}
+		}
+	}
+}
+
```

Comment:
Shouldn't this be inside `#ifdef CONFIG_PM ... #endif` or even `CONFIG_PM_DEVICE`?
And the same for their consequent calls.

Reply:
Code is guarded by `(IS_ENABLED(CONFIG_PM))`.
This way of doing allows to call `spi_stm32_pm_policy_state_lock_get` w/o `#ifdef CONFIG_PM` around, which helps gaining in readability.

Reply:
> Code is guarded by (IS_ENABLED(CONFIG_PM)).

yes

> which helps gaining in readability

but I don't think this is that important to justify the increase in code size when PM is disabled.
Lot of the other drivers don't include PM code by default.